inventory.go

```diff
@@ -1,23 +1,48 @@
 package main
 
 import (
-	"ansible-rackhd-inventory/rackhd"
 	"encoding/json"
 	"fmt"
 	"github.com/alexflint/go-arg"
+	"github.com/mlimaloureiro/ansible-rackhd-inventory/rackhd"
 	"github.com/spf13/viper"
 )
 
 type props struct {
 	rackhdUrl string
 	groups    []string
+	filterGroup string
 }
 
 type CLIArgs struct {
 	List bool
+	Host string
 }
 
 func main() {
+	props := getPropsFromConfig()
+	args := CLIArgs{}
+	arg.MustParse(&args)
+	// If argument "--list" was set to true
+	if args.List {
+		output ,err := handleList(props)
+		if err != nil {
+			panic(fmt.Errorf("Fatal error handling list: %s \n", err))
+		}
+		marshalResult, _ := json.MarshalIndent(output, "", "  ")
+		fmt.Println(string(marshalResult))
+		// If argument "--host" was set instead of host.
+	} else if args.Host != "" {
```

Comment:
instead of doing `else if` follow two rules:

- avoid switch cases / else ifs
- always return as soon as possible

this way it'll be easier to read the code. the sooner you leave the function, the easier it is to understand her purpose, even if it has different logics. in these case for instance

```
if args.List {
  // is doing something but doesn't return
} 
```

it will force you to continue reading the function to understand if any variable / logic inside the block is still mutating

preferably you should:

```
if args.List {
   // do something
   return
}
```